### PR TITLE
azurefog Keymap for Ergodox

### PR DESCRIPTION
**Pull request by azurefog**
## Description
RGB Matrix layer indicators
LEAD key usage
TAP_DANCE stuffs

### CODE REVIEW

File: layouts/community/ergodox/azurefog/config.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+  Set any config.h overrides for your specific keymap here.
+  See config.h options at https://docs.qmk.fm/#/config_options?id=the-configh-file
+*/
+#define DISABLE_RGB_MATRIX_DIGITAL_RAIN
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/config.h]: THis should include ... well, an include guard;
```suggestion
#pragma once 

#define DISABLE_RGB_MATRIX_DIGITAL_RAIN
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 2]
```diff
@@ -0,0 +1,385 @@
+#include "ergodox_ez.h"
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/keymap.c]: Could you replace this:
```suggestion
#include QMK_KEYBOARD_H
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+#include "debug.h"
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/keymap.c]: Could you remove all of these additional includes?  

```suggestion
```

They get included automatically, at various points in the built process, so they are redundant and not necessary here.

File: layouts/community/ergodox/azurefog/keymap.c [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+#include "action_layer.h"
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/keymap.c]:
```suggestion
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+#include "version.h"
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/keymap.c]:
```suggestion
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+
+
+enum custom_keycodes {
+  PLACEHOLDER = SAFE_RANGE, // can always be here
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/keymap.c]: You don't actually need a placeholder here.  It's absolutely fine being:
```suggestion
  RGB_SLD= SAFE_RANGE, // can always be here
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+
+void bs_dance (qk_tap_dance_state_t *state, void *user_data) {
+  switch (state->count) {
+    case 0 ... 1:
+      tap_code(KC_BSPACE);
+      break;
+    case 2:
+      register_code(KC_LSHIFT);
+      tap_code(KC_HOME);
+      unregister_code(KC_LSHIFT);
```
Comment (vomindoraan) [inline layouts/community/ergodox/azurefog/keymap.c]: You could replace these three lines with
```c
tap_code16(LSFT(KC_HOME));
```
It essentially does the same thing, except it's shorter and cleaner.

File: layouts/community/ergodox/azurefog/keymap.c [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+      tap_code(KC_DEL);
+      break;
+    case 3:
+      register_code(KC_LSHIFT);
+      register_code(KC_LCTRL);
+      tap_code(KC_HOME);
+      unregister_code(KC_LCTRL);
+      unregister_code(KC_LSHIFT);
```
Comment (vomindoraan) [inline layouts/community/ergodox/azurefog/keymap.c]:
```c
tap_code16(LSFT(LCTL(KC_HOME)));
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+      tap_code(KC_DEL);
+      break;
+    case 4:
+      register_code(KC_LSHIFT);
+      tap_code(KC_END);
+
+      unregister_code(KC_LSHIFT);
```
Comment (vomindoraan) [inline layouts/community/ergodox/azurefog/keymap.c]:
```c
tap_code16(LSFT(KC_END));
```

File: layouts/community/ergodox/azurefog/keymap.c [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+      tap_code(KC_DEL);
+      break;
+    case 5:
+      register_code(KC_LSHIFT);
+      register_code(KC_LCTRL);
+      tap_code(KC_END);
+      unregister_code(KC_LCTRL);
+      unregister_code(KC_LSHIFT);
```
Comment (vomindoraan) [inline layouts/community/ergodox/azurefog/keymap.c]:
```c
tap_code16(LSFT(LCTL(KC_END)));
```

File: layouts/community/ergodox/azurefog/rules.mk [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# Set any rules.mk overrides for your specific keymap here.
+# See rules at https://docs.qmk.fm/#/config_options?id=the-rulesmk-file
+SRC = matrix.c
```
Comment (drashna) [inline layouts/community/ergodox/azurefog/rules.mk]: This isn't actually needed.

```suggestion
```

**Comment by drashna** [comment]
Any updates on this?

**Comment by drashna** [comment]
It's been a few months without any update on this. 

I'm closing this for now.   If you wish to update this, please feel free to re-open the PR or resubmit it.

**Pull request closed without merging**